src/Microsoft.IdentityModel.Xml/DsigSerializer.cs

```diff
@@ -101,37 +101,42 @@ public virtual KeyInfo ReadKeyInfo(XmlReader reader)
                     // <X509Data>
                     if (reader.IsStartElement(XmlSignatureConstants.Elements.X509Data, XmlSignatureConstants.Namespace))
                     {
+                        var certificateData = new List<string>();
```

Comment:
if you create an X509Data you don't need these local variables. You can assign as you go.

Reply:
Done!